coderd/database/migrations/000274_check_tags.up.sql

```diff
@@ -0,0 +1,16 @@
+-- We need this as a type alias to ensure that we can override it in sqlc.
+-- We need to override it in sqlc so that it can be recognised as a StringMap.
+-- Without this zero values for other inferred types cause json syntax errors.
+CREATE DOMAIN tags AS jsonb;
+
+CREATE OR REPLACE FUNCTION tags_compatible(subset_tags tags, superset_tags tags)
```

Comment:
Since this is very specific for provisioners, perhaps the function could be called `provisioner_tags_contains(provisioner_tags, job_tags)` (swapped arg order as it matches contains functions in Go [haystack, needle]).

My thinking is that the renamed args don't expose business logic to the caller (i.e. who gives the subset and who the superset, etc).

Reply:
Could we take inspiration from our favourite testing library?

https://pkg.go.dev/github.com/stretchr/testify@v1.9.0/assert#Subset

Suggest:

```
tags_subset(superset tags, subset tags)
```

Reply:
I'm settling on a compromise between both of those suggestions and going for `tagset_contains`.

Reply:
Naming sounds fine, but I'm still strongly of the opinion that this function should be prefixed since it contains provisioner specific business logic (empty tag assertion).

Reply:
Updated.